tasks/create-advisory-internal-request/create-advisory-internal-request.yaml

```diff
@@ -0,0 +1,78 @@
+---
+apiVersion: tekton.dev/v1
+kind: Task
+metadata:
+  name: create-advisory-internal-request
+  labels:
+    app.kubernetes.io/version: "0.1.0"
+  annotations:
+    tekton.dev/pipelines.minVersion: "0.12.1"
+    tekton.dev/tags: release
+spec:
+  description: >-
+    Tekton task to create an advisory via an InternalRequest
+  params:
+    - name: jsonKey
+      type: string
+      description: The json key containing the advisory data
+      default: ".advisory"
+    - name: releasePlanAdmissionPath
+      type: string
+      description: Path to the JSON string of the ReleasePlanAdmission in the data workspace
+      default: "release_plan_admission.json"
+    - name: snapshotPath
+      type: string
+      description: Path to the JSON string of the Snapshot spec in the data workspace
+      default: "snapshot_spec.json"
+    - name: dataPath
+      type: string
+      description: Path to the data JSON in the data workspace
+      default: "data.json"
+    - name: request
+      type: string
+      description: Type of request to be created
+      default: "create-advisory"
+    - name: synchronously
+      type: string
+      description: Whether the task should wait for InternalRequests to complete
+      default: "true"
+  workspaces:
+    - name: data
+      description: Workspace where the json files are stored
+  steps:
+    - name: run-script
+      image: quay.io/hacbs-release/release-utils:6e92a6f8df8ef1cbecfb4c25b73ec6d92bded527
+      script: |
+        #!/bin/sh
+        #
+        #
+        set -e
+
+        # Obtain application from snapshot
+        application=$(jq -rc .application $(workspaces.data.path)/$(params.snapshotPath))
+
+        # Obtain origin workspace from releasePlanAdmission
+        origin=$(jq -rc '.spec.origin' $(workspaces.data.path)/$(params.releasePlanAdmissionPath))
+
+        # Extract the advisory key from the data JSON file
+        advisoryData=$(jq -rc "$(params.jsonKey)" $(workspaces.data.path)/$(params.dataPath))
+
+        echo -en "  Creating InternalRequest to create advisory..."
+        internal-request -r "$(params.request)" \
+                         -p application=${application} \
+                         -p origin=${origin} \
+                         -p advisory_json="${advisoryData}" \
+                         -s "$(params.synchronously)" \
+                         > $(workspaces.data.path)/ir-result.txt || \
+                         (grep "^\[" $(workspaces.data.path)/ir-result.txt | jq . && exit 1)
+        
+        internalRequest=$(awk 'NR==1{ print $2 }' $(workspaces.data.path)/ir-result.txt | xargs)
+        echo "done (${internalRequest})"
+
+        results=$(kubectl get internalrequest $internalRequest -o=jsonpath='{.status.results}')
+        if [ $(jq -jr '.result' <<< "${results}") == "Success" ]; then
```

Comment:
If the internal request fails, this script will already exit on line 67, no?

Reply:
The internal request should never fail per https://github.com/hacbs-release/app-interface-deployments/pull/55#pullrequestreview-1687553135

Reply:
I meant if it has a failed status. But since my previous commit I learnt that the use case here is for async - I didn't realize that. And same as I commented today on Leandro's PR, it actually won't work as intended. 

I assume that the purpose of this line below is so that if sync mode was used and the script detected a failure and returned a non-zero code, we will check for the status snippet in the output and if it's there, we fail right here. But the grep result has no effect on the outcome.

`(grep "^\[" $(workspaces.data.path)/ir-result.txt | jq . && exit 1)`

```sh
$ grep bla bla|jq . ; echo $?
grep: bla: No such file or directory
0
```

That's because jq will happily accept an empty input. If you wanted unsuccessful grep to fail it, you would need `set -o pipefail`.

Anyway, I think we would be better off doing different stuff based on the sync parameter.